docker-compose.yml

```diff
@@ -4,6 +4,11 @@
 
 version: '2'
 services:
+  build-requirements:
+    image: python:3.9
+    volumes:
+      - ./:/app
+    command: sh -c "cd /app && pip install pip --upgrade && pip install pip-tools && python -m piptools compile requirements.in --generate-hashes --allow-unsafe"
```

Comment:
Opted to put a comment instead, since the full Dockerfile is quite bloated to be used for this purpose, but we can revisit in the future to optimize a little more.